foundation/plugins/org.jboss.tools.foundation.ui/src/org/jboss/tools/foundation/ui/jobs/DisableAllWidgetsJob.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+  * Copyright (c) 2016 Red Hat, Inc.
+  * Distributed under license by Red Hat, Inc. All rights reserved.
+  * This program is made available under the terms of the
+  * Eclipse Public License v1.0 which accompanies this distribution,
+  * and is available at http://www.eclipse.org/legal/epl-v10.html
+  *
+  * Contributor:
+  *     Red Hat, Inc. - initial API and implementation
+  ******************************************************************************/
+package org.jboss.tools.foundation.ui.jobs;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.swt.graphics.Cursor;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.ui.progress.UIJob;
+import org.jboss.tools.foundation.ui.widget.WidgetUtility;
+
+public class DisableAllWidgetsJob extends UIJob {
```

Comment:
no javadoc. agreement was all public classes in foundation should have good docs and tests.